### PR TITLE
Continue updating PRs if a merge fails

### PR DESCRIPTION
**Pull request by anarast**
This might need some more testing/another test case - let me know if you have any ideas

### CODE REVIEW

**Comment by chinthakagodawita** [comment]
Nice one, I'll give it a squiz this weekend. Probably need this to be logged as an `error` too, not `info`.

**Comment by chinthakagodawita** [comment]
@anarast could you take a look? Added a more stringent test: [`ba4c102`](https://github.com/chinthakagodawita/autoupdate/pull/102/commits/ba4c102811a5c0111088771b2c609323ee53df0e)

File: test/autoupdate.test.ts [hunk 1]
```diff
@@ -604,4 +604,77 @@ describe('test `merge`', () => {
 
     expect(scope.isDone()).toEqual(true);
   });
+
+  test('continue if merging throws an error', async () => {
```
Comment (anarast) [inline test/autoupdate.test.ts:608]: this is what i wanted to do! i couldn't figure out how to make multiple scopes work

File: src/autoupdater.ts [hunk 2]
```diff
@@ -122,7 +122,15 @@ export class AutoUpdater {
...
     }
 
-    await this.merge(mergeOpts);
+    try {
+      await this.merge(mergeOpts);
+    } catch (e) {
+      ghCore.error(
+        `Caught error running merge, skipping and continuing with remaining PRs`,
+      );
+      ghCore.setFailed(e);
```
Comment (anarast) [inline src/autoupdater.ts:131]: is this necessary if we're already logging out the error in the merge method?

Comment (chinthakagodawita) [reply src/autoupdater.ts:131]: Yea - `setFailed` sets the process exitcode which marks this run as failed (which in turn sends an email to the repository / PR owner)

Comment (anarast) [reply src/autoupdater.ts:131]: cool lgtm

**Pull request merged**